src/openrct2/world/SpriteBase.h

```diff
@@ -33,6 +33,16 @@ struct SpriteBase
     uint8_t sprite_direction;
 
     void MoveTo(const CoordsXYZ& newLocation);
+    template<typename T> bool Is() const;
+    template<typename T> T* As()
+    {
+        T* result = nullptr;
+        if (Is<T>())
+        {
+            result = reinterpret_cast<T*>(this);
+        }
+        return result;
```

Comment:
Personally I'd just use ternary, but this is a personal preference
```suggestion
        return Is<T>() ? reinterpret_cast<T*>(this) : result;
```